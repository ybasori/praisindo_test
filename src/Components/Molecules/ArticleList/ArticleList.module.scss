.article {
  margin-top: 16px;
}
.load-more {
  width: 100%;
  border: 1px solid #000;
  box-sizing: border-box;
  padding: 16px;
  text-align: center;
  margin-top: 16px;
  cursor: pointer;
}
.loading {
  width: 100%;
  box-sizing: border-box;
  text-align: center;
}
