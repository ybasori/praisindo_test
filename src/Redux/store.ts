import { configureStore } from "@reduxjs/toolkit";
import nyTimesReducer from "./NyTimes/NyTimes.reducer";

export const store = configureStore({
  reducer: {
    nyTimes: nyTimesReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ serializableCheck: false }),
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
