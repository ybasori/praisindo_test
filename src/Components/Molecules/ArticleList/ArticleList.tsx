import React, { useEffect } from "react";
import { AppDispatch, RootState } from "../../../Redux/store";
import { useDispatch, useSelector } from "react-redux";
import { getSearchArticles } from "../../../Redux/NyTimes/NyTimes.thunk";
import styles from "./ArticleList.module.scss";
import ArticleItem from "../../Atoms/ArticleItem/ArticleItem";
import { loadMore } from "../../../Redux/NyTimes/NyTimes.reducer";

const ArticleList = () => {
  const dispatch: AppDispatch = useDispatch();
  const nyTimesState = useSelector((state: RootState) => state.nyTimes);

  useEffect(() => {
    dispatch(
      getSearchArticles({
        keyword: nyTimesState.keyword,
        page: nyTimesState.page,
      })
    );
  }, [dispatch, nyTimesState.keyword, nyTimesState.page]);

  const onLoadMore = () => {
    dispatch(loadMore());
  };

  return (
    <div className={styles["container"]}>
      {nyTimesState.loadingArticles && nyTimesState.articles === null ? (
        <div className={styles["loading"]}>
          <i className="fas fa-circle-notch fa-spin"></i>
        </div>
      ) : (
        <>
          {nyTimesState.articles && nyTimesState.articles.length > 0 ? (
            <>
              {nyTimesState.articles.map((item, index) => (
                <div key={index} className={styles["article"]}>
                  <ArticleItem
                    title={item.abstract}
                    author={item.byline.original}
                    publishDate={item.pub_date}
                    headline={item.lead_paragraph}
                  />
                </div>
              ))}
              <div
                className={`${styles["load-more"]} ${styles["is-loading"]}`}
                onClick={() =>
                  nyTimesState.loadingArticles ? null : onLoadMore()
                }
              >
                {nyTimesState.loadingArticles ? (
                  <i className="fas fa-circle-notch fa-spin"></i>
                ) : (
                  "Load More"
                )}
              </div>
            </>
          ) : (
            <>No Data</>
          )}
        </>
      )}
    </div>
  );
};

export default ArticleList;
