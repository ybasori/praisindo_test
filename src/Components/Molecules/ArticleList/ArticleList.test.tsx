import { render } from "@testing-library/react";
import ArticleList from "./ArticleList";
import { store } from "../../../Redux/store";
import { Provider } from "react-redux";

jest.mock("axios", () => ({
  data: { response: { docs: [] } },
}));

describe("Test ArticleList", () => {
  it("Snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <ArticleList />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
