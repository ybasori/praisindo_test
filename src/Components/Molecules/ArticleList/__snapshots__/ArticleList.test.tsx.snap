// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ArticleList Snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="loading"
    >
      <i
        class="fas fa-circle-notch fa-spin"
      />
    </div>
  </div>
</div>
`;
